freqtrade/strategy/strategy_helper.py

```diff
@@ -83,3 +97,98 @@ def stoploss_from_open(open_relative_stop: float, current_profit: float) -> floa
+    :param current_rate: Current asset price.
+    :return: Positive stop loss value relative to current price
+    """
+    return 1 - (stop_rate / current_rate)
+
+
+def informative(timeframe: str, asset: Optional[str] = None,
+                fmt: Optional[Union[str, Callable[[KwArg(str)], str]]] = None,
+                ffill: bool = True) -> Callable[[Callable[[Any, DataFrame, dict], DataFrame]],
+                                                Callable[[Any, DataFrame, dict], DataFrame]]:
+    """
+    A decorator for populate_indicators_Nn(self, dataframe, metadata), allowing these functions to
+    define informative indicators.
+
+    Example usage:
+
+        @informative('1h')
+        def populate_indicators_1h(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
+            dataframe['rsi'] = ta.RSI(dataframe, timeperiod=14)
+            return dataframe
+
+    :param timeframe: Informative timeframe. Must always be higher than strategy timeframe.
+    :param asset: Informative asset, for example BTC, BTC/USDT, ETH/BTC. Do not specify to use
+    current pair.
+    :param fmt: Column format (str) or column formatter (callable(name, asset, timeframe)). When not
+    specified, defaults to {asset}_{name}_{timeframe} if asset is specified, or {name}_{timeframe}
+    otherwise.
+    * {asset}: name of informative asset, provided in lower-case, with / replaced with _. Stake
+    currency is not included in this string.
+    * {name}: user-specified dataframe column name.
+    * {timeframe}: informative timeframe.
+    :param ffill: ffill dataframe after mering informative pair.
+    """
+    def decorator(fn: Callable[[Any, DataFrame, dict], DataFrame]):
+        @wraps(fn)
+        def wrapper(self, dataframe: DataFrame, metadata: dict) -> DataFrame:
+            nonlocal timeframe, asset, fmt
+            # Modifying variables inherited from parent scope poisons other wrapper instances!
+            asset_c = asset or ''
+            fmt_c = fmt
+
+            # Default format.
+            if fmt_c is None:
+                fmt_c = '{name}_{timeframe}'
+                if asset_c:
+                    fmt_c = '{asset}_' + fmt_c
+
+            if asset_c:
+                # Insert stake currency if needed.
+                asset_c = self._format_pair(asset_c)
+            else:
+                # Not specifying an asset will define informative dataframe for current pair.
+                asset_c = metadata['pair']
+
+            inf_metadata = {'pair': asset_c, 'timeframe': timeframe}
+            inf_dataframe = self.dp.get_pair_dataframe(asset_c, timeframe)
+            inf_dataframe = fn(self, inf_dataframe, inf_metadata)
+
+            # Clear stake currency from specified asset name if it is a spot pair.
+            asset_short = asset_c
+            if '/' in asset_c:
+                asset_short = re.sub(rf'([^/])(/{self.config["stake_currency"]})?', r'\1',
+                                     asset_c)
```

Comment:
I'm not quite sure i understand the intend with this, but i think it's to counter the side-effects from trying to be clever and only keeping one side of the pair.


as mentioned elsewhere - i'd always keep both sides of the pair in the name. 
using only one will become confusing and may end up with colliding names (assume i use 1h - BTC/USDT and BTC/BUSD ).

Reply:
I am optimizing for most common case. 99% of strategies will not use a sibling stake currency, so whats the point to impose this extra typing on majority of users? Besides in such case user would get `btc_rsi_1h` + `btc_busd_rsi_1h`. IMO this little asymmetry in a very rare case beats users having to constantly hammer in stake currency. It also solves hardcoding of stake currency in strategy code. If we always kept stake currency then suddenly most common usecase would warrant a strategy modification when switching from USDT to BUSD as column names would need changing.

Reply:
Please see above - where i posted a sample where changing stake-currency will require changes in the remainder due to this.